src/docfx/restore/DirectoryPackage.cs

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Enumeration;
+
+namespace Microsoft.Docs.Build
+{
+    internal abstract class DirectoryPackage : Package
```

Comment:
DirPackage {
string _dir;
Package _package;
}

Package CreateSub(dir) => new DIrPackage(this, dir);